tools/snp.sh

```diff
@@ -904,13 +918,203 @@ attest_guest() {
+  sudo subscription-manager repos --enable codeready-builder-for-rhel-9-x86_64-rpms
+
+  # qemu dependencies
+  sudo dnf install -y ninja-build
+  sudo dnf install -y pkg-config
+  sudo dnf install  -y glib2-devel
+  sudo dnf install  -y pixman-devel
+  sudo dnf install -y libslirp-devel
+   
+  # ovmf dependencies
+  sudo dnf install -y uuid-devel
+  sudo dnf install -y iasl
+  sudo dnf remove nasm
+  install_nasm_from_source
+
+  # kernel dependencies
+  sudo dnf install -y rsync
+  sudo dnf install -y ncurses-devel
+  
+  # libssl-dev is openssl-devel in RHEL
+  # rpm-build -- Scripts and executable programs used to build packages
+  sudo dnf install -y rpm-build
+
+  # cloud-utils dependency
+  sudo dnf install -y cloud-init
+  
+  # sev-snp-measure
+  sudo dnf install -y python3-pip
+
+  install_common_dependencies
+  echo "true" > "${dependencies_installed_file}"
+}
+
+fedora_install_dependencies() {
+  local dependencies_installed_file="${WORKING_DIR}/dependencies_already_installed"
+  
+  check_if_dependencies_installed
+  # If dependencies already exist
+  if [[ $dependencies_installed -eq 1 ]]; then
+    return 0
+  fi
+  
+  # Build dependencies
+  sudo dnf update
+  sudo dnf install -y git make ninja-build gcc glib2 glib2-devel pixman pixman-devel meson
+  
+  # For fix of FAILED: libqemuutil.a.p/util_async.c.o
+  # Uncomment wget command for applying patch d66ba6dc for async.c as per https://gitlab.com/qemu-project/qemu/-/issues/1655
+  # wget "https://raw.githubusercontent.com/qemu/qemu/d66ba6dc1cce914673bd8a89fca30a7715ea70d1/util/async.c" -O "${SETUP_WORKING_DIR}/AMDSEV/qemu/util/async.c"
+
+  sudo dnf install -y libuuid libuuid-devel
+  sudo dnf install -y python
+
+  # For fixing Failed to execute command make tbuild
+  sudo dnf remove nasm
+  install_nasm_from_source
+  sudo dnf install -y acpica-tools zstd rpm-build dwarves perl
+  
+  # # cloud-utils dependency
+  # sudo dnf install -y cloud-init
+  
+  # # sev-snp-measure
+  # sudo dnf install -y python3-pip
+
+  # install_common_dependencies
+  # echo "true" > "${dependencies_installed_file}"
+}
+
+ubuntu_install_dependencies() {
+  local dependencies_installed_file="${WORKING_DIR}/dependencies_already_installed"
+  
+  check_if_dependencies_installed
+
+  # If dependencies already exist
+  if [ $dependencies_installed -eq 1 ]; then
+    return 0
+  fi
+
+  # Build dependencies
+  sudo apt install -y build-essential git
+  
+  # qemu dependencies
+  sudo apt install -y ninja-build pkg-config
+  sudo apt install -y libglib2.0-dev
+  sudo apt install -y libpixman-1-dev
+  sudo apt install -y libslirp-dev
+  
+  # ovmf dependencies
+  sudo apt install -y python-is-python3 uuid-dev iasl
+  
+  #sudo apt install -y nasm
+  sudo apt purge nasm
+  install_nasm_from_source
+
+  # kernel dependencies
+  sudo apt install -y bc rsync
+  sudo apt install -y flex bison libncurses-dev libssl-dev libelf-dev dwarves zstd debhelper
```

Comment:
I feel it'd be better to split this up into multiple lines for the sake of uniformity